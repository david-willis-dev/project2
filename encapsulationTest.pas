program main;
    type ExampleClass = class
        private
            key: integer;
       end;

    var x: ExampleClass;

begin
  x := ExampleClass;
  writeln('Trying to access private variable... ');
  x.key := 90;
end.