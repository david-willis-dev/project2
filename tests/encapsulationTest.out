Trying to access private variable...
Hey! You can't access that variable here. tried to access: x.key from out of its appropriate context. Encapsulation successfully protected :)